extending/template_system/search_table.rst

```diff
@@ -0,0 +1,125 @@
+.. _template_search_table:
+
+
+HTML output for :command:`conan search` table
+=============================================
+
+.. warning::
+
+    This has to be an considered as an **experimental** feature, we might
+    change the context provided to this templates once we have more exmpales
+    from the community.
+
+
+The :ref:`conan_search` command can generate an HTML table with the results of the
+query when looking for binaries
+
+.. image:: ../../images/conan-search_binary_table.png
+    :height: 500 px
+    :width: 600 px
+    :align: center
+
+
+This is the default Canan provides, but you can use your own Jinja2 template to customize
+this output to your needs:
+
+ * *<cache>/templates/output/search_table.html*.
+ 
+
+Context
+-------
+
+Conan feeds this template with the information about the packages found, this information
+is called context and it contains these objects:
+
+ * ``base_template_path``: path to the directory where the choosen template file is
+   located. It is needed if your output file needs to link assets distributed together
+   with the template file.
+ * ``search``: it contains the pattern used in the command line to search packages.
+ * ``results``: this object contains all the information retrieved from the remotes, it
+   is used to get the headers and the rows.
+
+
+When the output is a table, the first thing needed are the headers, these can be a sinle row
+or two rows like the image above. In order to get the headers you should use
+``results.get_headers(keys)`` with a list of extra ``keys`` you want to include. Conan will
+always return a header for all the different settings and options values, with this ``keys``
+you can retrieve other information that might be useful in your table like ``remote``, 
+``reference``, ``outdated`` or ``package_id``.
+
+Then you can use the returned job to get the actual headers:
+
+ * single row headers: it just returns a list with all the headers, it is straightforward to use:
+
+   .. code-block:: html
+
+        <thead>
+            <tr>
+                {%- for header in headers.row(n_rows=1) %}
+                <th>{{ header }}</th>
+                {%- endfor %}
+            </tr>
+        </thead>
+            
+ * two-rows headers: it returns a list of tuples like the following one:
```

Comment:
I am not sure I follow. list of tuples is python object, right? but template logic seems to be called inside the html file. do you mean this will result in tempalte preprocessor inserting string representation of the `list of tuples` here?

Reply:
In Jinja2 templates the variables are **python objects**, not just strings. As such python objects you can execute methods on them. In the example above a line was missing, it should be:

```html
        <thead>
            <tr>
                {%- set headers = results.get_headers(keys=['remote', 'package_id']) %}
                {%- for header in headers.row(n_rows=1) %}
                <th>{{ header }}</th>
                {%- endfor %}
            </tr>
        </thead>
```

Here `headers` is an object, `headers.row(n_rows=1)` returns an actual iterable and inside the template you iterate the items as you would do in Python itself (lot of Jinja2 syntax sugar here, but it is like python itself).

Reply:
I know this *two-rows header* thing is not the easiest thing, neither the HTML code to compose the headers of the table from this python object (see all those loops, colspan, rowspan,...)